### DISCUSSION TITLE
монстры, мобы

### DISCUSSION THREAD

**Discussion by Pigzinzspace** [godot-engine-ru/shooter-3d#9] [Ideas]
**содержание документа** текст содержит перечень всех монстров в игре, монстры разделены по биомам, описывается их внешний вид, описывает их поведение и анимации как это выглядит 
перечень монстров по биомам 
шаблон
**название биома** "название локаций в которых встречается через запятую"
**название монстра** 
внешний вид:
"описание как выглядит моб"
поведение:
"описание атаки или атак, описание перемещения как он движется гонится за игроком, поведения вне боя, описание как умирает" 

**офис/технический/научный сектор**  
**человек недавно ставший зомби**
внешний вид: 
люди в белых халатах, люди в синих робах люди в костюмах, в строгих женских платьях, зомби в форме охраны, зомби техничка в переднике
поведение: 
стоит в неестественной "сломанной" позе, сидит или полу лежит, ест труп сидя на корточках откусывая от тела опираясь на руки как животное, реагирует на игрока внимательным разворотом головы, бежит к игроку с постоянным ускорением (разгоняясь) на 2-ух ногах или встав на четвереньки, руки болтаются как палки (не сгибаются), с загнутыми кистями или вытянуты вперед с растопыренными пальцами.  

**цех**
**здоровяк**
внешний вид:
роба рабочего или джинсы голый торс или майка или футболка или рубашка с закатанными рукавами, тело может быть с  татуировкой, такая  же женская фигура.   Раздутое тело и лицо покрытое вздутостями,  волдырями и гнойниками, издает низкие рычащие звуки,  или хрипение. 
поведение: медленно бьет в стену двумя сжатыми в кулаки руками, бьется головой об стену или колонну, стоит оперевшись плечом или спиной на стену, бежит на игрока с ревом и рыканьем иногда опираясь на передние конечности как это делают обезьяны, хватает рядом стоящего или бегущего зомби  метает его как снаряд  в игрока этот бросок наносит урон, доставленный броском зомби не умирает, а начинает атаковать, атакует в ближнем бою серией из множества  ударов в одну точку, может схватить игрока в захват и перекинуть его через себя (игроку приходится вставать, он не может двигаться и стрелять).      

**секретная лаборатория**
**зомби лягушка**
внешний вид:
сильно деформированное  человеческое тело без одежды, тело раздуто похоже на  подушку с шипастыми хитиновыми наростами, тонкие и длинные для человека конечности  с когтями   вместо пальцев, руки не отличаются от ног, передвигается  на 4-ех лапах, пока не двигается сливается с окружающим, но не до конца заметить можно. 
поведение:
сидит неподвижно обычно в углах помещения, иногда на стенах или на потолке,  передвигается прыжками, атакует из далека, плюется кислотой или гноем, не подходит, при приближение к ней отпрыгивает, назад,  во время боя часто воет  призывая других мобов, которые через 3 4 секунды спавнятся на входах в помещение.  

**внутренний двор**
**Босс - объект-127**
внешний вид:
зомби здоровяк из цеха, в бронированном экзоскелете, с ракетными касетами на плечах
поведение:
если игрок на дистанции дальнего боя стреляет в него ракетами или может метать зомби как это делает здоровяк, если на дистанции ближнего боя то атакует одной из атак: таран, бросок через себя или серия молотящих ударов.

 **секретная лаборатория**
 **шипованный зомби**
 внешний вид:
 зомби в драной военной форме, у которого к рукам и груди прикреплены металлические пластины с шипами, на уровне кистей рук торчат несколько крючьев.
 поведение: 
 ведет себя как обычный зомби но на ближней дистанции хватает, обнимает игрока и наносит ему урон, от него невозможно отцепиться. после смерти отваливается нанося еще немного урона.   

**встречается везде**
**токсичный зомби**
внешний вид:
желтого цвета, без одежды, окруженный желто-зеленой подсветкой,  в темноте светится ярким зеленым светом. 
поведение: 
ударив игрока в мили или получив урон, блюет ему под ноги зеленую лужу, которая наносит урон и пузырится, тоже происходит когда зомби  умирает. Лужа остается на 1 минуту.     

**анимация критических попаданий**
 зомби отстреливают или отбивают плечо и ключицу остается прямая линия от шеи до ребер, из раны хлещет фонтан тело у водит в сторону и начинает вихлять, потом падает
 взрывается голова труп оседает на колени и продолжает сидеть, кровь волной как через край выливается на грудь и спину
 взрывается голова, из нее хлещет кровь, труп продолжает бежать вперед через пару секунд разворачивается и бежит под прямым углом к прежнему курсу
 отрывается нога, зомби катится вперед как будто бегущему поставили подножку

**Comment by alex7850** [comment]
Опасаюсь, что лягуха будет раздражать своим призывом. Куда бы интереснее смотрелось захватывание языком, как у курильщика в l4d или чтобы слизь приманивала зомбей.

Механику вставания после ранения (аля left4dead) я бы подробнее описал еще где-то

**Reply by Pigzinzspace** [reply]
ты имеешь в виду после раняния?
герой видит перед собой пол, нажимаешь пробел тебе показывают руку перед лицом и как она отталкивает тебя от пола
но это анимация героя это скорее в тему куклы героя.
а если про зомби то они летят расставив руки и ноги как тряпки но если попадают остаются на ногах, а если пролетают мимо то сперва на четвереньки у них есть анимация на четвереньках
по жабе анимация контроля в шутанах когда тебя замедляют или хватают языком бесит обычно гораздо сильнее чем призыв мобов. ну и это не будет выглядеть что они с неба летят она зовет из соседних комнат пару обычных калек на пару ударов кувалды

**Comment by slapin** [comment]
Предлагаю старых добрых слаймов, их очень просто анимировать, проще делать боссами и простор для творчества неограничен.

**Reply by me2beats** [reply]
вроде на зомбях уже остановились

**Reply by Pigzinzspace** [reply]
типо зомби превращаются в желе и собираются в слаймов как второй терминатор, а потом трансфорсируются в пустых
![image](https://user-images.githubusercontent.com/79694583/151705849-3d9e21b4-1fc3-4d3c-b8e0-4e75266712bc.png)

**Comment by Einshtein12** [comment]
Предлагаю зомби-кота: обитает везде, ведёт себя как агрессивный кот. После убиймтва и оживления становиться помощником см. мой комментарий в #11, выглядит примерно так https://myloview.ru/poster-cat-zombie-cartoon-pet-green-monster-vector-illustration-no-D205D67

**Reply by Pigzinzspace** [reply]
убийства  и оживлениея ?

**Reply by Einshtein12** [reply]
Да но оживление оч дорогостоящее, и оживить можно определённое кол-во существ(например 5)
